packages/lab/src/avatar/Avatar.tsx

```diff
@@ -1,50 +1,88 @@
-import { FlexLayout, makePrefixer } from "@salt-ds/core";
+import { makePrefixer } from "@salt-ds/core";
 import { UserIcon } from "@salt-ds/icons";
 import { clsx } from "clsx";
 import { forwardRef, HTMLAttributes, ReactNode } from "react";
+import { useLoaded } from "./internal/useLoaded";
 
 import "./Avatar.css";
 
+export type InitialsGetter = (name: string) => string | undefined;
+
 export interface AvatarProps extends HTMLAttributes<HTMLDivElement> {
-  children?: ReactNode | string;
+  /**
+   * The name that Avatar represents.
+   */
+  name?: string;
+  /**
+   * Defines the function that gets initials. Default is capital first letter of each separate word in name.
+   * If a function is not passed or returns undefined, Avatar will default to Icon.
+   */
+  initialsGetter?: InitialsGetter;
+  /**
+   * Image src of Avatar.
+   */
+  src?: string;
+  /**
+   * Multiplier for the base avatar.
+   */
   size?: number;
+  /**
+   * Icon to be used as a default item. Defaults to `UserIcon`
+   */
   fallbackIcon?: ReactNode;
 }
 
 const withBaseName = makePrefixer("saltAvatar");
 const DEFAULT_AVATAR_SIZE = 2; // medium
 
+const initialsDefaultGetter = (name: string | undefined) =>
+  name &&
+  name
+    .split(" ")
+    .map((n) => n[0])
+    .join("")
+    .toUpperCase();
```

Comment:
should we limit the amount of initials? or add an ellipsis, dash etc on overflow? (e.g. 10 word name shows the middle initials only when it overflows)